enhancements/network/user-defined-network-segmentation.md

```diff
@@ -415,6 +415,7 @@ The CRDs spec defines as follows:
 | ExcludeSubnets | List of CIDRs.<br/>IP addresses are removed from the assignable IP address pool and are never passed to the pods.                                                                                                                                                                                                                                                                                      | Yes      |
 | JoinSubnets    | Subnet used inside the OVN network topology.  When omitted, this means no opinion and the platform is left to choose a reasonable default which is subject to change over time.                                                                                                                                                                                                                        | Yes      |
 | IPAMLifecycle  | Control IP addresses management lifecycle. When `Persistent` is specified it enable workloads have persistent IP addresses. For example: Virtual Machines will have the same IP addresses along their lifecycle (stop, start migration, reboots). Supported by Topology `Layer2` & `Localnet`.                                                                                                         | Yes      |
+| IPAMMode       | Control how much of the IP configuration will be managed by OVN-Kubernetes. Must be one of `Full`, `Disabled`.                                                                                                                                                                                                                                                                                         | Yes      |
```

Comment:
Is it too late to introduce IPAM type for aggragating IPAM configurations? 
For example:
```
ipam: 
  mode: [Disabled/Full]
  lifecycle: [Presistent]
```

Reply:
yeah I was thinking about that, but not sure if we want to make even more changes so close to the release, let's ask on the team meeting

Reply:
IMHO it **does** make sense to aggregate IPAM into a type. 

Nice suggestion.

Reply:
It could also make sense to aggregate subnets & joinSubents as well:
L2 primary example:
```
spec:
  topology: Layer2
  layer2:
    role: Primary
    ipam:
      mode: <Disabled/Full>
      lifecycle: <Presistent>
      subents: <> (mandatory for primary)
      joinSubents: <> (optional)
    mtu: <..> (otional)   
    
```
Note that this change should require more work (adjust the code, unit-tests, and tests - u/s and d/s).